sources/models/src/modeled_types/ecs.rs

```diff
@@ -291,3 +291,61 @@ mod test_ecs_agent_image_pull_behavior {
         }
     }
 }
+
+/// ECSDurationValue represents a string that contains a valid ECS duration value
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ECSDurationValue {
+    inner: String,
+}
+
+lazy_static! {
+    pub(crate) static ref ECS_DURATION_VALUE: Regex =
+        Regex::new(r"^(([0-9]+\.)?[0-9]+h)?(([0-9]+\.)?[0-9]+m)?(([0-9]+\.)?[0-9]+s)?$").unwrap();
```

Comment:
Curious how this regex was derived? If we're going off [the Go `time.ParseDuration` API](https://pkg.go.dev/time#ParseDuration) (where this duration string will be consumed):

> A duration string is a possibly signed sequence of decimal numbers, each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".

This doesn't support negative time durations (like `-1h` with the leading sign) which probably makes sense for this use case but would be a valid duration to be consumed by the Go serialization. I think we should support it just in case customers are doing something with negative time duration (and if the ECS agent supports it)

---

[The Go source code](https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/time/format.go;l=1523) does offer a regex we could reference:
```
[-+]?([0-9]*(\.[0-9]*)?[a-z]+)+
```
although this regex is more permissive of the unit types. Instead, [they reference a static in memory unit slice and valid units that way.](https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/time/format.go;l=1506-1515)

Reply:
> . I think we should support it just in case customers are doing something with negative time duration (and if the ECS agent supports it)

Even thought the deserialization of negative values is supported in the golang API, the minimum duration value supported by the ECS agent is `1s`, hence the regex. 

> I think we should support it just in case customers are doing something with negative time duration (and if the ECS agent supports it)

I thought on supporting the same regex provided by the golang APIs, but decided against that because it seems odd to let a user set `-1h` in the API system, and then have the ECS agent ignore the value due to internal validations. I think the interface contract is between the API and the ECS agent, not the API with golang primitives, thus the API system should only allow what the ECS agent considers "valid" values, instead of what the golang APIs allow.

Reply:
That makes total sense - thanks for the deep dive!! 👏